worker/addfmusite/addFMUSite.py

```diff
@@ -0,0 +1,99 @@
+#  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+#  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+#  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+#  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+########################################################################################################################
+
+from __future__ import print_function
+import sys
+import os
+import boto3
+import json
+import tarfile
+import shutil
+import time
+from subprocess import call
+import logging
+
+if len(sys.argv) == 3:
+    fmu_upload_name = sys.argv[1]
+    upload_id = sys.argv[2]
+else:
+    print('addSite called with incorrect number of arguments: %s.' % len(sys.argv), file=sys.stderr)
+    sys.exit(1)
+
+if not upload_id:
+    print('upload_id is empty', file=sys.stderr)
+    sys.exit(1)
+
+directory = os.path.join('/parse', upload_id)
+
+try:
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+except:
+    print('error making add site parsing directory for upload_id: %s' % upload_id, file=sys.stderr)
+    sys.exit(1)
+
+s3 = boto3.resource('s3', region_name='us-east-1', endpoint_url=os.environ['S3_URL'])
+key = "uploads/%s/%s" % (upload_id, fmu_upload_name)
+# fmu files gets uploaded with user defined names, but here we rename
+# to model.fmu to avoid keeping track of the (unreliable, non unique) user upload name
+# createFMUTags will however use the original fmu upload name for the display name of the site
+fmupath = os.path.join(directory, 'model.fmu')
+jsonpath = os.path.join(directory, 'tags.json')
+
+bucket = s3.Bucket('alfalfa')
+bucket.download_file(key, fmupath)
+
+call(['python', 'addfmusite/createFMUTags.py', fmupath, fmu_upload_name, jsonpath])
+
+site_ref = False
+with open(jsonpath) as json_file:
+    data = json.load(json_file)
+    for entity in data:
+        if 'site' in entity:
+            if entity['site'] == 'm:':
+                site_ref = entity['id'].replace('r:', '')
+                break
+
+if site_ref:
+    # This adds a new haystack site to the database
+    call(['npm', 'run', 'start', jsonpath, site_ref])
+
+    # Open the json file and get a site reference
+    # Store the results by site ref
+    def reset(tarinfo):
+        tarinfo.uid = tarinfo.gid = 0
+        tarinfo.uname = tarinfo.gname = "root"
+        return tarinfo
+
+    tarname = "%s.tar.gz" % site_ref
+    tar = tarfile.open(tarname, "w:gz")
+    tar.add(directory, filter=reset, arcname=site_ref)
+    tar.close()
+
+    bucket.upload_file(tarname, "parsed/%s" % tarname)
+    
+    #os.remove(tarname)
```

Comment:
@goldcoder01  since we are uploading back to s3, go ahead and remove all of the files so that the worker does not fill up